packages/api/src/user/user.ts

```diff
@@ -73,10 +78,180 @@ export const createPlatformUser = async (
+      reject(err)
+    }
+  })
+}
+
+export const autoLogin = async (
+  apiUrl: string,
+  autoLoginToken: string,
+  loginChallenge: string,
+  cookies: string,
+): Promise<string> => {
+  const resp = await axios.post<RedirectToResponse>(
+    `${apiUrl}/v2/oauth2/login?token=${autoLoginToken}&login_challenge=${loginChallenge}`,
+    { withcredentials: true, headers: { Cookie: cookies } },
+  )
+
+  return resp.data.redirect_to
+}
+
+const mustGetRedirectUrl = (resp: http.IncomingMessage): string => {
+  const location = resp.headers.location
+  if (!location) {
+    throw new Error('No redirect location found')
+  }
+
+  return location
+}
+
+const mustGetCookies = (resp: http.IncomingMessage): string[] => {
+  const cookies = resp.headers['set-cookie']
+
+  if (!cookies) {
+    console.log(resp.headers)
+    throw new Error(
+      `no cookies found for request ${resp} with redirect ${location}`,
+    )
+  }
+
+  return cookies
+}
+
+const mustGetParam = (url: string, param: string): string => {
+  const value = new URLSearchParams(new URL(url).searchParams).get(param)
+
+  if (!value) {
+    throw new Error('no query parameter found with name: ' + param)
+  }
+
+  return value
+}
+
+const joinCookies = (cookies: string[]) => cookies.join('; ')
+
+const provideConsent = async (
+  apiUrl: string,
+  consentChallenge: string,
+  cookies: string,
+): Promise<string> => {
+  const { data } = await axios.post<RedirectToResponse>(
+    `${apiUrl}/v2/oauth2/consent?consent_challenge=${consentChallenge}`,
+    {
+      withcredentials: true,
+      headers: { Cookie: cookies },
+      data: { isRejected: false },
+    },
+  )
+
+  return data.redirect_to
+}
+
+export const loginPlatformUser = async (
+  client: Client,
+  authSdk: auth.SDK,
+  clientSdk: auth.ClientSDK,
+  userID: string,
+  redirectUri: string,
+): Promise<RequestTokenResponseData> => {
+  const autoLoginToken = await getUserMasqueradeToken(client, {
+    userId: userID,
+  })
+
+  const [loginUrl, codeVerifier] = await authSdk.getLogInUrl()
+
+  let cookies: string[] = []
+
+  const initLoginRedirect = await haltRedirect(loginUrl)
+
+  const loginRedirect = mustGetRedirectUrl(initLoginRedirect)
+  cookies = cookies.concat(mustGetCookies(initLoginRedirect))
+  const loginChallenge = mustGetParam(loginRedirect, 'login_challenge')
+
+  const loginRedirectUrl = await autoLogin(
+    client.getBaseUrl(),
+    autoLoginToken.token,
+    loginChallenge,
+    joinCookies(cookies),
+  )
+
+  const initConsentResp = await haltRedirect(
+    loginRedirectUrl,
+    joinCookies(cookies),
+  )
+  const consentRedirect = mustGetRedirectUrl(initConsentResp)
+  cookies = cookies.concat(mustGetCookies(initConsentResp))
+  const consentChallenge = mustGetParam(consentRedirect, 'consent_challenge')
+
+  const consentRedirectUrl = await provideConsent(
+    client.getBaseUrl(),
+    consentChallenge,
+    joinCookies(cookies),
+  )
+
+  const initExchangeCode = await haltRedirect(
```

Comment:
Why `init`?

```suggestion
  const exchangeCodeResponse = await haltRedirect(
```

Reply:
using `init` here along with some of the other requests to signify that its a request that would typically be redirected, eg we are `initializing` the exchange code request flow, just as we `initialize` the consent flow.

I've updated to `initExchangeCodeResp`